app/routes/login.tsx

```diff
@@ -11,17 +11,23 @@ import {
 } from '@chakra-ui/react';
 import { LockIcon } from '@chakra-ui/icons';
 import type { ActionArgs } from '@remix-run/node';
+import { redirect } from '@remix-run/node';
 import { Form } from '@remix-run/react';
 
-import { createUserSession } from '~/session.server';
+import { getUsername } from '~/session.server';
+import { createLoginRequest } from '~/saml.server';
 
 export const action = async ({ request }: ActionArgs) => {
-  return createUserSession({
-    request: request,
-    username: 'starchartdev',
-    remember: false,
-    redirectTo: '/',
-  });
+  // Check if a session with a username exists
+  const user = await getUsername(request);
+
+  // If not then create a login request to the IDP's redirect binding
+  if (!user) {
+    const samlRedirectURL = await createLoginRequest(new URL(request.url));
```

Comment:
Question: what is `request.url` going to be here?

What about the case of `http://localhost:8080/login?redirectTo=%2Fdev`, where I want to go to `/dev` but get redirected to the login page?